mlflow/projects/utils.py

```diff
@@ -53,13 +53,18 @@ def _parse_subdirectory(uri):
     subdirectory = ""
     parsed_uri = uri
     if "#" in uri:
-        subdirectory = uri[uri.find("#") + 1 :]
-        parsed_uri = uri[: uri.find("#")]
+        subdirectory = _fix_quote(uri[uri.find("#") + 1 :])
+        parsed_uri = _fix_quote(uri[: uri.find("#")])
     if subdirectory and "." in subdirectory:
         raise ExecutionException("'.' is not allowed in project subdirectory paths.")
     return parsed_uri, subdirectory
 
 
+def _fix_quote(uri):
+    # Remove quotes and, if necessary, add new ones
+    return quote(uri.strip("'\""))
```

Comment:
@dinaldoap Thanks for the PR! I'm wondering why we need `quote` here.

Reply:
@harupy Thanks for the feedback! You question made me think in a better solution after reviewing mlflow's code. Since parsed_uri and subdirectory aren't used in command lines, they don't need to be quoted again. Besides that, I observed that removing quotes from uri instead of parsed_uri and subdirectory will allow "mlflow run" to work with quoted URIs without subdirectories. For example, mlflow run "'exam ples/sklearn_elasticnet_wine/'".
I will rewrite my commits in my branch and force push them in a minute.